### ISSUE TITLE
Track and catch regressions in micro benchmarks

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#2947]
We have several micro benchmarks, e.g. `perf_simple_query`, `perf_fast_forward`, `perf_cache_eviction`. We should have a CI environment which will store results of those tests and allow us to catch regressions.

**Comment by slivne** [comment]
@argenet - will start with the tests related to #2981

**Comment by mykaul** [comment]
@yaronkaikov - I think you are implementing this these days, no?

**Comment by yaronkaikov** [comment]
@mykaul not exactly, since we changed the way we run those tests (https://github.com/scylladb/scylladb/commit/d661d03), I am fixing it so it will run again. If we need to add more tests i can add it, i just need to know what

**Comment by roydahan** [comment]
All of the above already running for years and results are being stored, but recently @avikivity said we need to run them using RAM drive to reduce disk noise.

@yaronkaikov do you have the task/issue to create the ram drive as @avikivity suggested on monster?

**Comment by yaronkaikov** [comment]
@roydahan we have one now https://github.com/scylladb/scylla-pkg/issues/3311.  i guess this can be closed

**Comment by mykaul** [comment]
> Hm, I get a 404 when trying to access [scylladb/scylla-pkg#3311](https://github.com/scylladb/scylla-pkg/issues/3311) I'm curious how you all track your benchmarks. I've been working on a continuous benchmarking tool, Bencher: https://github.com/bencherdev/bencher And I've been trying to research how others solve the same problem.

I'm not sure I understand the license of bencher (especially that plus directory...), so haven't looked too closely at it.
Personally, I've used in the past https://github.com/tobami/codespeed/ (see https://speed.pypy.org/ for a usecase), but it doesn't seem maintained anymore.

I've also worked with the team who created and maintains https://distributed-system-analysis.github.io/pbench/ 

The above issue is in a private repo. It's just the test framework to run the test (against a RAM disk). I assume we'll run it as part of a Jenkins job and use xunit to store the results... Unsure yet.

**Comment by mykaul** [comment]
> Thanks for the pointers, @mykaul ! Both `codespeed` and `pbench` are new to me. I'll definitely take a look.
> 
> As for the Bencher license, it is a more permissive variation of the GitLab license: https://gitlab.com/gitlab-org/gitlab/-/blob/master/LICENSE
> 
> The idea is that the enterprise and licensing features live under a directory that is source available, and the rest of the code base is open source (Apache 2.0/MIT). Do you have any advice on how I can better explain that/make things more clear?

It's complex. In ScyllaDB for example, we have different repos, for enterprise and OSS.

**Comment by epompeii** [comment]
> I assume we'll run it as part of a Jenkins job and use xunit to store the results... Unsure yet.

If you are interested in using Bencher, I would be up for adding `xunit` support.
This would allow you to both store the results, graph them, and use statistical thresholds to catch performance regressions in CI.
